src/transformers/tokenization_utils_base.py

```diff
@@ -728,7 +728,7 @@ def as_tensor(value, dtype=None):
                     value = [value]
 
                 if not is_tensor(value):
-                    tensor = as_tensor(value)
+                    tensor = as_tensor(np.array(value))
```

Comment:
The `as_tensor` (defined a few line) above actually deals with the case where each element has different length.
So a directly `np.array(value)` is not going to work with all the case, as shown in the failing CI.